src/osp/OneStepProverHostIo.sol

```diff
@@ -254,49 +305,45 @@ contract OneStepProverHostIo is IOneStepProver {
 
             // NOTE we are expecting the following layout for our proof data, similar
             // to that expected for the point evaluation precompile
-            // [:32] - versionhash (eigenlayer)
+            // [:32] - hash (eigenlayer) (not versioned like 4844)
             // [32:64] - evaluation point
             // [64:96] - expected output
             // [96:224] - g2TauMinusG2z
             // [224:288] - kzg commitment (g1 point)
             // [288:352] - proof (g1 point)
             // [352:385] - preimage length
-
-            // expect first 32 bytes of proof to be the expected version hash
-            require(bytes32(kzgProof[:32]) == leafContents, "KZG_PROOF_WRONG_HASH");
-
+            
             {
-                // evaluation point
-                uint256 evaluationPoint = uint256(bytes32(kzgProof[32:64]));
+                
+                uint256[2] memory kzgCommitment = [uint256(bytes32(kzgProof[224:256])), uint256(bytes32(kzgProof[256:288]))];
+                uint256[4] memory alphaMinusG2 = [uint256(bytes32(kzgProof[96:128])), uint256(bytes32(kzgProof[128:160])), uint256(bytes32(kzgProof[160:192])), uint256(bytes32(kzgProof[192:224]))];
+                uint256[2] memory proofUint256 = [uint256(bytes32(kzgProof[288:320])), uint256(bytes32(kzgProof[320:352]))];
+                uint256 z = uint256(bytes32(kzgProof[32:64]));
+                uint256 y = uint256(bytes32(kzgProof[64:96]));
 
-                // expected output
-                uint256 expectedOutput = uint256(bytes32(kzgProof[64:96]));
+                require(kzgCommitment[0] < BN254.FP_MODULUS, "COMMIT_X_LARGER_THAN_FIELD");
+                require(kzgCommitment[1] < BN254.FP_MODULUS, "COMMIT_Y_LARGER_THAN_FIELD");
 
-                BN254.G2Point memory g2TauMinusG2z = BN254.G2Point({
-                    X: [uint256(bytes32(kzgProof[96:128])), uint256(bytes32(kzgProof[128:160]))],
-                    Y: [uint256(bytes32(kzgProof[160:192])), uint256(bytes32(kzgProof[192:224]))]
-                });
+                require(proofUint256[0] < BN254.FP_MODULUS, "PROOF_X_LARGER_THAN_FIELD");
+                require(proofUint256[1] < BN254.FP_MODULUS, "PROOF_Y_LARGER_THAN_FIELD");
 
-                BN254.G1Point memory kzgCommitment =
-                    BN254.G1Point(uint256(bytes32(kzgProof[224:256])), uint256(bytes32(kzgProof[256:288])));
+                require(z < BN254.FR_MODULUS, "Z_LARGER_THAN_FIELD");
+                require(y < BN254.FR_MODULUS, "Y_LARGER_THAN_FIELD");
+
+                require((keccak256(kzgProof[224:288]) == bytes32(kzgProof[:32])) && (bytes32(kzgProof[:32]) == leafContents), "KZG_PROOF_WRONG_HASH");
```

Comment:
I remembered we hashed datalength into the data too. why is it correct here

Reply:
It appears that the place where we hash data and length isn't referenced here and seems to be isolated away. In this code it appears that the data for the msp tree is from the other way the hash is created. This is the same way for 4844 from what I can tell. Interesting part is, even if we match the hash data to each other, there is still no connection between the 2 systems on the contract level.